.github/scripts/cmd/validator/validator.go

```diff
@@ -0,0 +1,196 @@
+package validator
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/coinhall/yacar/internal/enums"
+	"github.com/coinhall/yacarsdk/v2"
+)
+
+func Start(filePaths []string) {
+	validateYacarJSONs(filePaths)
+	log.Println("Validated JSONs successfully...")
+}
+
+func validateYacarJSONs(filePaths []string) {
+	var wg sync.WaitGroup
+	for _, filePath := range filePaths {
+		wg.Add(1)
+		go func(filePath string) {
+			defer wg.Done()
+
+			file, err := os.Open(filePath)
+			if err != nil {
+				log.Fatalf("error while opening file: %s", err)
+			}
+			defer file.Close()
+
+			if err := validateYacarJSON(filePath, file); err != nil {
+				log.Fatalf("%s\npath: %s", err, filePath)
+			}
+
+		}(filePath)
+	}
+	wg.Wait()
+}
+
+func validateYacarJSON(filePath string, file *os.File) error {
+	switch {
+	case strings.Contains(filePath, enums.Account.Name()):
+		return validateAccountJSON(file)
+	case strings.Contains(filePath, enums.Asset.Name()):
+		return validateAssetJSON(file)
+	case strings.Contains(filePath, enums.Binary.Name()):
+		return validateBinaryJSON(file)
+	case strings.Contains(filePath, enums.Contract.Name()):
+		return validateContractJSON(file)
+	case strings.Contains(filePath, enums.Entity.Name()):
+		return validateEntityJSON(file)
+	case strings.Contains(filePath, enums.Pool.Name()):
+		return validatePoolJSON(file)
+	default:
+		return fmt.Errorf("unknown file path: %s", filePath)
+	}
+}
+
+func validateAccountJSON(file *os.File) error {
+	var accounts []yacarsdk.Account
+
+	if err := json.NewDecoder(file).Decode(&accounts); err != nil {
+		return fmt.Errorf("error while decoding JSON: %s", err)
+	}
+
+	idCount := make(map[string]bool)
```

Comment:
A better definition for a `set` is `map[string]struct{}` as it is slightly more optimized (struct{} takes 0 bytes while bool takes 4 bytes and is useless).